sdk/identity/identity/test/internal/node/managedIdentityCredential.spec.ts

```diff
@@ -232,12 +226,16 @@ describe("ManagedIdentityCredential", function() {
       ...mockHttpClient.tokenCredentialOptions
     });
 
-    const promise = credential.getToken("scopes");
+    // Sinon's clock has some issues with Node 16.
+    // If we remove this conditional on Node 16, we get the following error:
+    //   PromiseRejection HandledWarning: Promise rejection was handled asynchronously
+    // It will point to the `await promise` line below.
+    const clock = process.version.startsWith("v16") ? undefined : sandbox.useFakeTimers();
 
-    imdsMsiRetryConfig.startDelayInMs = 80; // 800ms / 10
+    const promise = credential.getToken("scopes");
 
-    // 800ms -> 1600ms -> 3200ms, results in 6400ms, / 10 = 640ms
-    clock.tick(640);
+    // 800ms -> 1600ms -> 3200ms, results in 6400ms
+    await clock?.tickAsync(6400);
```

Comment:
Are the numbers retry timeouts or something? Some context on what you're advancing would be good.

Reply:
This test checks that an error is thrown after we have a repeated number of responses with 404. The timeout comes from the exponential retry policy, that increases on each retry